spec/lexers/ecl_spec.rb

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+describe Rouge::Lexers::ECL do
+  let(:subject) { Rouge::Lexers::ECL.new }
+
+  describe 'guessing' do
+    include Support::Guessing
+
+    it 'guesses by filename' do
+      assert_guess :filename => 'foo.ecl'
+      assert_guess :filename => 'FOO.ECL'
+    end
+
+    it 'guesses by mimetype' do
+      assert_guess :mimetype => 'application/x-ecl'
+    end
+  end
+
+  describe 'lexing' do
+    include Support::Lexing
+
+    it 'recognizes one-line comments not followed by a newline (#796)' do
+      assert_tokens_equal '// comment', ['Comment.Single', '// comment']
+    end
```

Comment:
I suspect you will need full coverage of each category / pattern within the tests?

Reply:
We don't actually mind if the tests are a bit light. Comprehensively testing a syntax highlighter is pretty difficult so if you just want to use the visual sample to eyeball it, that's acceptable from the perspective of the maintainers.

Reply:
Don't worry. I am doing it more for our internal use. We submit more than one lexer and it would be helpful for our other lexers. I will push an update later.

Reply:
No worries. If you want an example of relatively thorough tests, [the GHC Core spec](https://github.com/rouge-ruby/rouge/blob/a821e3f2f5b135345ab51c14ae75002a39eb2cf3/spec/lexers/ghc_core_spec.rb) is pretty good.